Updated EKS blog (#5143)
pulumi/docs: content/blog/multi-lang-components/index.md

---

> title: "Use Amazon Elastic Kubernetes Service (EKS) in any Pulumi language"

Reviewer: "in any language supported by Pulumi"

Author: ```suggestion
title: "Use Amazon Elastic Kubernetes Service (EKS) in any language supported by Pulumi"
```

> title: "Use Amazon Elastic Kubernetes Service (EKS) in any language supported by Pulumi"

Reviewer: ```suggestion
title: "Create Amazon Elastic Kubernetes Service (EKS) clusters in any language supported by Pulumi"
```

---

Pulumi's infrastructure as code tooling combines the programming languages and tools you already know with the full power of cloud
infrastructure. Until now, some Pulumi components for cloud infrastructure, like our popular [EKS package] for Amazon's Elastic
Kubernetes Service, were only available in a subset of the languages supported by Pulumi.

Now, you can use the [EKS package]–previously only available for TypeScript–in all four Pulumi languages: TypeScript, Python, .NET,
and Go. Regardless of the language you choose, you can [manage EKS clusters] with Pulumi, starting with the [v0.22.0 release of the
EKS package]. Check out our Modern Infrastructure Wednesday video to see it in action:

{{< youtube "gxLyAr0lUg0?rel=0" >}}

<!--more-->

> EKS in all languages is made possible by our ongoing work to enable "[multi-language components]". With multi-language components,

Reviewer 2: We need a section explaining what components are, I think. Probably before this paragraph. Starting point/example:

> Pulumi's EKS module is an example of a Pulumi _component_. A component is a collection of Pulumi resources which are grouped together and can span multiple cloud providers. They provide an easy way to create cloud resources which follow best practices which can then be package up and consumed by Pulumi users

Author: ```suggestion
### Making components available in every language

Pulumi's EKS module is an example of a Pulumi component. A component is a collection of Pulumi resources that are grouped together and can span multiple cloud providers. Components provide an easy way to package best practices for using cloud resources. Until now, components could only be used in Pulumi programs of the same language as the component's language.

To change that, we're making "[multi-language components]". With multi-language components, 
```

> EKS in all languages is made possible by our ongoing work to enable "[multi-language components]". With multi-language components,

Reviewer: probably makes sense to add a header above this section that says something like "Making components available in every language"

any member of the worldwide Pulumi community will be able to author a Pulumi component in the language they're most comfortable

> with and share it with all Pulumi users: even those that use a different language than the component was authored in. Multi-language

Reviewer 2: I like the idea of adding something here that talks about the amplification effects of writing in one language and duplicating into other languages. Something like

"Before multi-language components, unless you were willing to write the code in all of our languages, you would only be able to reach Pulumi users using the same language SDK. With multi-language components, writing in one language will now reach 3x as language platforms and countless new users, with no extra code needed!"

components take care of the conversion from one language to other Pulumi languages, so component developers don't need detailed

> knowledge of the other languages.Multi-language components will also enable organizations to write Pulumi components in the right

Reviewer 2: ```suggestion
knowledge of the other languages. Multi-language components will also enable organizations to write Pulumi components in the right
```

> knowledge of the other languages.Multi-language components will also enable organizations to write Pulumi components in the right

Reviewer: ```suggestion
knowledge of the other languages. Multi-language components will also enable organizations to write Pulumi components in the right
```

> programming language, or languages, for each team and each product. An infrastructure or platform teams will be able to write

Reviewer: ```suggestion
programming language, or languages, for each team and each product. Infrastructure or platform teams will be able to write
```

> components in Go and share them with a webapp team using TypeScript, or write in .NET and share with a machine learning team using

Reviewer: ```suggestion
components in Go and share them with a webapp team using TypeScript, write components in .NET and share them with a machine learning team using
```

> Python, or any other combination that suits your business.

Reviewer: ```suggestion
Python, and any other combination that suits your business.
```

### Next steps

The next step in the story is a public preview of a seamless authoring experience that will enable component creators to write, build,

> and publish multi-language components. If you'd like, follow along with our progress:

Reviewer 2: ```suggestion
and publish multi-language components. We're excited to make the Pulumi components you create easier, more accessible and reach more cloud engineers!  If you'd like, follow along with our progress:
```

EKS package]. Check out our Modern Infrastructure Wednesday video to see it in action:

{{< youtube "gxLyAr0lUg0?rel=0" >}}

<!--more-->

### Making components available in every language

Pulumi's EKS package is made up of Pulumi components. A component is a collection of Pulumi [resources] that provides an easy way to
package best practices for using cloud resources, even if those best pratices span individual resources or even entire cloud providers.
Until now though, components could only be used in Pulumi programs written in the same language as the component.

To change that, we're creating "[multi-language components]". With multi-language components, anyone can author a component in one
language and automatically compile it to all other languages supported by Pulumi. This means anyone in the worldwide Pulumi community
can take advantage of any component, even if it's written in a different language. Before multi-language components, unless a component
was willing to write the code in all four Pulumi languages, their component would only be able to reach Pulumi users using the

> same language SDK. With multi-language components, that same component will now reach the other three language platforms and countless

Reviewer 2: I think there's been a merge issue here, looks like we're repeating ourselves

infrastructure. Until now, some Pulumi components for cloud infrastructure, like our popular [EKS package], were
only available in a subset of the languages supported by Pulumi.

Now, you can use the [EKS package]–previously only available for TypeScript–in all four Pulumi languages: TypeScript, Python, .NET,
and Go. Regardless of the language you choose, you can [manage EKS clusters] with Pulumi, starting with the [v0.22.0 release of the
EKS package]. Check out our Modern Infrastructure Wednesday video to see it in action:

{{< youtube "gxLyAr0lUg0?rel=0" >}}

<!--more-->

> EKS in all languages is made possible by our ongoing work to enable "multi-language components". With multi-language components,

Reviewer: should we link to this issue? https://github.com/pulumi/pulumi/issues/2430

Reviewer: probably should also mention that this is a feature in public preview for those who are not aware of it yet

Author: ```suggestion
EKS in all languages is made possible by our [ongoing work] to enable "multi-language components". With multi-language components,
```

> a developer can author a Pulumi component in their preferred language and then publish it for all Pulumi languages—without needing

Reviewer: I suggest writing this in future tense to indicate that the full functionality is not yet available. "a developer will be able to author..."

Author: ```suggestion
a developer will be able to author a Pulumi component in their preferred language and then publish it for all Pulumi languages—without needing
```

detailed knowledge of the other Pulumi languages.

> EKS in all languages is just the start of this story. Multi-language components will enable all organizations to choose

Reviewer: the overarching benefit we should weave into here is that this will facilitate easier sharing of components with the entire Pulumi ecosystem of developers who are using different languages

Author: Adjusted

the right programming language, or languages, for each team and each product. Your infrastructure or platform teams will be able to
write components in Go and share them with a webapp team using TypeScript, or write in .NET and share with a machine learning team
using Python or any other combination that suits your business. Similarly, any member of the worldwide Pulumi community will be able to
author a component in the language they're most comfortable with and share it with all Pulumi users.

The next step in the story is a seamless authoring experience that enables component creators to write, build, and publish components

> without detailed knowledge of the various Pulumi languages. We're excited for you to try it soon - check back here for updates!

Reviewer: we should end with a CTA. I suggest leading people to the eks package and maybe also link to the multi language components issue so that they can follow along

Author: Great idea, done!

> The next step in the story is a seamless authoring experience that enables component creators to write, build, and publish components
> without detailed knowledge of the various Pulumi languages. We're excited for you to try it soon - check back here for updates!

Author: ```suggestion
The next step in the story is a seamless authoring experience that enables component creators to write, build, and publish components
without detailed knowledge of the various Pulumi languages. We're excited for you to try it soon - check back here for updates!

### Next steps

- Get started with the new pulumi-eks package  https://www.pulumi.com/blog/easily-create-and-manage-aws-eks-kubernetes-clusters-with-pulumi/
- Subscribe to the [multi-language components issue] for updates on the public preview
```

infrastructure. Until now, some Pulumi components for cloud infrastructure, like our popular [EKS package], were
only available in a subset of the languages supported by Pulumi.

Now, you can use the [EKS package]–previously only available for TypeScript–in all four Pulumi languages: TypeScript, Python, .NET,
and Go. Regardless of the language you choose, you can [manage EKS clusters] with Pulumi, starting with the [v0.22.0 release of the
EKS package]. Check out our Modern Infrastructure Wednesday video to see it in action:

{{< youtube "gxLyAr0lUg0?rel=0" >}}

<!--more-->

> EKS in all languages is made possible by our ongoing work to enable "[multi-language components]". With multi-language components,

Reviewer: ```suggestion
EKS in all languages is made possible by our ongoing work to enable "[multi-language components]," which is in public preview. With multi-language components,
```

any member of the worldwide Pulumi community will be able to author a Pulumi component in the language they're most comfortable
with and share it with all Pulumi users: even those that use a different language than the component was authored in. Multi-language
components take care of the conversion from one language to other Pulumi languages, so component developers don't need detailed
knowledge of the other languages.Multi-language components will also enable organizations to write Pulumi components in the right
programming language, or languages, for each team and each product. An infrastructure or platform teams will be able to write
components in Go and share them with a webapp team using TypeScript, or write in .NET and share with a machine learning team using
Python, or any other combination that suits your business.

### Next steps

> The next step in the story is a public preview of a seamless authoring experience that enables component creators to write, build,

Reviewer: ```suggestion
We are hard at work to release the next phase of this feature into the public preview: a seamless authoring experience that enables component creators to write, build,
```

Reviewer: "that will enable component creators to write, build,..."

Author: ```suggestion
The next step in the story is a public preview of a seamless authoring experience that will enable component creators to write, build,
```
